### ISSUE TITLE
Purchases.purchasePackage() not opening the OS purchase window after subscription expired.

### ISSUE THREAD

**Issue by dpedrinha** [RevenueCat/purchases-flutter#201]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-flutter/blob/develop/CONTRIBUTING.md)
- [x] I have read the [Help Center](https://support.revenuecat.com/hc/en-us)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-flutter/issues)

**Describe the bug**
Here's the scenario:
I install the app and make an auto renewable subscription purchase (iOS).
I wait for it to expire.
I uninstall the app, install back again and try to purchase that same subscription with `Purchases.purchasePackage()`.
**problem**
It doesn't open the device purchase window. Instead, it just returns `PurchaserInfo `with `isActive = false` and info (dates) regarding the first purchase, including the date when it expired.

If I try to restore the purchase, it says there are not active subscriptions, which is correct.

It should open the purchase window when the purchase is not active.

I'm running flutter on master channel v 2.3.0-17.0.pre.375 and purchases-flutter v 3.3.0

**Comment by aboedo** [comment]
hi! thanks for reporting! I'll try to reproduce on my side to see what's going on. 

In the meantime, are you using StoreKit config files by any chance?

**Comment by dpedrinha** [comment]
> In the meantime, are you using StoreKit config files by any chance?

Negative.

**Comment by Niko-by** [comment]
I am seeing the same problem. At the same time, this was not the case in older versions of the plugin, in my other projects it works well because there is an old version of the plugin. Did you find a way to fix this?

**Comment by vegaro** [comment]
I have tried with our own example in a 14.6 physical iPhone and I can't seem to reproduce it. What I am doing is:

- Purchase a monthly package.
- Go to settings and cancel the renewal.
- Wait for it to expire.
- Reopen the app. Try to purchase.
- I can see the purchase dialog just fine.

I am curious about something. Is there anything different between our sample (https://github.com/RevenueCat/purchases-flutter/tree/develop/revenuecat_examples/MagicWeather) and your project? Like in the way the plugin is configured or the purchases are initiated. Can you try the sample project on your device and see if you can reproduce it? The plugin version would need to be updated to match yours (I believe 3.3.0) in that case.

Thanks!

**Comment by Niko-by** [comment]
> I have tried with our own example in a 14.6 physical iPhone and I can't seem to reproduce it. What I am doing is:
> 
> * Purchase a monthly package.
> * Go to settings and cancel the renewal.
> * Wait for it to expire.
> * Reopen the app. Try to purchase.
> * I can see the purchase dialog just fine.
> 
> I am curious about something. Is there anything different between our sample (https://github.com/RevenueCat/purchases-flutter/tree/develop/revenuecat_examples/MagicWeather) and your project? Like in the way the plugin is configured or the purchases are initiated. Can you try the sample project on your device and see if you can reproduce it? The plugin version would need to be updated to match yours (I believe 3.3.0) in that case.
> 
> Thanks!

Here are my steps:
1. All subscriptions are in the same subscription group
2. The application has not been published, it is in alpha testing
3. I have successfully completed my first subscription
4. Then I try to (upgrade / downgrade) it to another subscription, and in response I receive a message about successful payment, but the subscription window did not even appear. I attached a log below.
5. I click again and try to increase / decrease the subscription, and only now the dialogue appears, I successfully subscribe but it has not changed. When I looked at the contents of PurchaseInfo, I found that both activeSubscriptions subscriptions remained there: [10_vacancy_1_month, 5_vacancy_1_month]. But the subscription had to be renewed and replaced with one another. Why didn't it work?


My logs:

[Purchases] - DEBUG: ℹ️ Products request finished.
[Purchases] - DEBUG: 💰 Retrieved SKProducts:
[Purchases] - DEBUG: 💰 5_vacancy_1_month - <SKProduct: 0x28217eed0>
[Purchases] - DEBUG: ℹ️ 1 completion handlers waiting on products
[Purchases] - DEBUG: ℹ️ makePurchase
[Purchases] - DEBUG: 💰 Purchasing product - 5_vacancy_1_month
[Purchases] - DEBUG: ℹ️ PaymentQueue updatedTransaction: 5_vacancy_1_month 1000000829856984 ((null)) 1000000827816553 - 1
[Purchases] - DEBUG: ℹ️ Loaded receipt from url file:///private/var/mobile/Containers/Data/Application/0F5344C3-1951-4CB1-A5A6-F7501157643A/StoreKit/sandboxReceipt
[Purchases] - DEBUG: ℹ️ Found 0 unsynced attributes for App User ID: $RCAnonymousID:8d17eb05aaa14d7081c93cb98410272d
[Purchases] - DEBUG: ℹ️ There are no requests currently running, starting request POST /receipts
[Purchases] - DEBUG: ℹ️ API request started: POST /v1/receipts
8.0.0 - [Firebase/Analytics][I-ACS023141] Purchase is a duplicate and will not be reported. Product ID: 5_vacancy_1_month
[Purchases] - DEBUG: ℹ️ API request completed with status: POST /v1/receipts 200
[Purchases] - DEBUG: ℹ️ Serial request done: POST /receipts, 0 requests left in the queue
[Purchases] - DEBUG: 💰 Finishing transaction 5_vacancy_1_month 1000000829856984 (1000000827816553)
[GETX] CLOSE DIALOG: 997658559
flutter: delivery product 5_vacancy_1_month

**Comment by Niko-by** [comment]
I am using the function - await Purchases.purchaseProduct (kId)

**Comment by vegaro** [comment]
It looks like your problem is different to the original post. @Niko-by you're having issues while upgrading/downgrading. @dpedrinha you're having issues when repurchasing the same product after expiration. Am I correct?

**Comment by vegaro** [comment]
I see @Niko-by you also commented in https://github.com/RevenueCat/purchases-flutter/issues/202 which looks more similar to what you're experiencing. I will look into that one and see if I can reproduce it.

**Comment by Niko-by** [comment]
> It looks like your problem is different to the original post. @Niko-by you're having issues while upgrading/downgrading. @dpedrinha you're having issues when repurchasing the same product after expiration. Am I correct?

I think these are similar problems, since in both cases, the subscription window does not open.

**Comment by vegaro** [comment]
@Niko-by I tried upgrading from a monthly to an annual subscription and it's showing the dialog in my project using 3.3.0 and an iOS 14.6 device. 

We haven't really changed anything from 3.2.2 to 3.3.0 that could cause this. We just added a new function unrelated to purchases. I saw in the other issue that your other projects are working fine. Have you tried using the same version of the plugin in your new project and in your other working projects? Or have you tried upgrading the version of the plugin in your other projects and confirmed that upgrading the version breaks it?

**Comment by vegaro** [comment]
By the way, are you getting any exception? Have you tried adding a try/catch to the purchaseProduct call? If the call is throwing a PlatformException, we might be able to pinpoint the source of the problem

**Comment by Niko-by** [comment]
I have 3 subscriptions and they are all with a 1 month period. For example silver, gold, platinum. I need to switch from gold to platinum, they have different prices, but the same term (1 month). Maybe this is the problem? How i can do it?

And yes, I have added an exception in the code, but there is no error there. Apparently from the logs that everything is fine.

Here is a portion of my code:

await Purchases.purchaseProduct(
      kId,
      type: type,
      upgradeInfo: activeTarif.value != freeTarif
          ? UpgradeInfo(activeTarif.value,
              prorationMode: ProrationMode.immediateWithTimeProration)
          : null,
    ).then((value) async {
      await Purchases.syncPurchases();
      deliverProduct(kId);
      updateActiveTarif();
      Get.back();
      print('Purchase success');
    }).catchError((error) {
      Get.back();
      if (error.code != '1') {
        showError(error.message.toString());
      }
      print('Error purchase $error');
    });

**Comment by Niko-by** [comment]
In other applications, I had 1 subscription but with different periods (1 month, 3 months and 1 year). There, the tariff change works fine. But in the current project, I need to change them between different tariffs.

**Comment by vegaro** [comment]
I am seeing the same behavior as you. When I purchase another subscription in the same subscription group, Apple doesn't offer the upgrade, and instead it treats it as a different subscription.

I have to say that upgrades/crossgrades have never properly worked on sandbox, but they do work fine in production. I am confident to say this is a quirk of sandbox, and totally unrelated to RevenueCat. We have seen all sort of issues with sandbox, sometimes they get fixed for a while before they get broken again. This is also mentioned in the *Important Takeaways* section of this blog post https://www.revenuecat.com/blog/the-ultimate-guide-to-subscription-testing-on-ios

Fwiw, I found a post in the Apple developers forum describing the same issue https://developer.apple.com/forums/thread/680087

It looks like in your other apps you have different subscription groups. The behavior is different in that case https://www.revenuecat.com/blog/the-ultimate-guide-to-subscription-testing-on-ios. Changing subscription groups is called crossgrade, which also used to be broken in sandbox.

**Comment by Niko-by** [comment]
Please tell me how to change the subscription correctly? Because on android, I get the same behavior, when I change the tariff, I have 2 tariffs left to hang at once.

When buying, there is an upgradeInfo parameter, what exactly should be transferred there? Can you show me an example? I sent the product ID, but, apparently, something else needs to be substituted there. I didn't find this in your examples.

**Comment by Niko-by** [comment]
Already figured out, it works fine on android. I will check it on iOS after release. Thanks for the help.

**Comment by dpedrinha** [comment]
@vegaro sorry for the late reply. I had to deal with another problem with Apple for a published app.

I can't test the sample. To be honest, I am not allowed to clone it to my computer and I don't know how to fork it, so I had to check the files online. And it's specially hard to do it when the file names and class names are not the same or even related. It makes no sense to me, but anyways.

As far as I understood, I'm doing the same. 

As I said before, I already made a purchase with this same user on this same device. Uninstalled the app. And now, every time I run the app and try to make a purchase I can't, because `Purchases.purchasePackage(_offerings.current.monthly);` doesn't open the native purchase window (on iOS).

Here're the steps I'm taking with the results I'm getting:

```
Future<void> _initPlatformState() async {
    await Purchases.setup("******************************", appUserId: null);
    Purchases.addPurchaserInfoUpdateListener((info) {
      _checkIfVIP(info);
    });
    try {
      purchaserInfo = await Purchases.getPurchaserInfo();
      print('PurchaserInfo: ' + purchaserInfo.toString());
      //_checkIfVIP(purchaserInfo);
    } on PlatformException catch (e) {
      print('PlatformException: ' + e.toString());
    }
    print('is user vip? ' + isVIP.toString());
  }

  void _checkIfVIP(PurchaserInfo purchaserInfo) {
    if (purchaserInfo != null && purchaserInfo.entitlements.all['vip'] != null) {
      setIsVIP(purchaserInfo.entitlements.all['vip'].isActive);
    } else {
      setIsVIP(false);
    }
    notifyListeners();
  }

  Offerings _offerings;

  Future<Offerings> getOfferings() async {
    Offerings offerings;
    if (_offerings != null)
      offerings = _offerings;
    else {
      try {
        offerings = await Purchases.getOfferings();
      } on PlatformException catch (e) {
        print(e.toString());
      }
    }
    print(offerings.toString());
    return offerings;
  }
```

Here I start the app by calling `_initPlatformState()` and I get a non null but empty `purchaserInfo`.

Moving forward, the user eventually get's to the purchases screen with a button that calls:

```
Future<void> doPurchase() async {
    try {
      if (_offerings != null) {
        _toggleLoadingState();
        PurchaserInfo purchaserInfo = await Purchases.purchasePackage(_offerings.current.monthly);
        // the native purchase window should open before going forward
        if (purchaserInfo.entitlements.all["vip"].isActive) {
          MainModel mainModel = Provider.of<MainModel>(context, listen: false);
          mainModel.setIsVIP(true);
          showDialog(
            context: context,
            builder: (BuildContext context) {
              return AlertDialog(
                title: new Text(S.current.welcome_to_the_vip_group),
                content: new Text(
                  S.current.purchase_success,
                  textAlign: TextAlign.center,
                ),
                actions: <Widget>[
                  new TextButton(
                    child: new Text(S.current.close),
                    onPressed: () => {popToPreviousPage(context)},
                  ),
                ],
              );
            },
          );
        } else {
          print('purchaserInfo.entitlements.all["vip"].toString());
          _toggleLoadingState();
        }
      }
    } on PlatformException catch (e) {
      _toggleLoadingState();
      var errorCode = PurchasesErrorHelper.getErrorCode(e);
      showDialog(
        context: context,
        builder: (BuildContext context) {
          if (errorCode == PurchasesErrorCode.purchaseCancelledError) {
            return AlertDialog(
              title: new Text(''),
              content: new Text(
                S.current.purchase_canceled_error,
                textAlign: TextAlign.center,
              ),
              actions: <Widget>[
                new TextButton(
                  child: new Text(S.current.close),
                  onPressed: () => Navigator.pop(context),
                ),
              ],
            );
          } else if (errorCode == PurchasesErrorCode.storeProblemError) {
            return AlertDialog(
              title: new Text(S.current.oops),
              content: new Text(S.current.purchase_store_problem_error(Platform.isAndroid ? 'Google Play' : 'App Store')),
              actions: <Widget>[
                new TextButton(
                  child: new Text(S.current.close),
                  onPressed: () => Navigator.pop(context),
                ),
              ],
            );
          } else {
            return AlertDialog(
              title: new Text(S.current.oops),
              content: new Text(S.current.purchase_default_error),
              actions: <Widget>[
                new TextButton(
                  child: new Text(S.current.close),
                  onPressed: () => Navigator.pop(context),
                ),
              ],
            );
          }
        },
      );
      print('isLoadingSomething = ' + e.toString());
    }
  }
```

Here, when I call `PurchaserInfo purchaserInfo = await Purchases.purchasePackage(_offerings.current.monthly);` it should open the native purchase screen before going forward, but instead, it immediately responds with a non null non empty `purchaserInfo` with `emptypurchaserInfo.entitlements.all["vip"].isActive == false` (because it expired long ago). 

And that's where I'm stuck.

**Comment by dpedrinha** [comment]
And here's the debug log after I try `doPurchase()`

> [Purchases] - DEBUG: ℹ️ Vending Offerings from cache
> [Purchases] - DEBUG: ℹ️ makePurchase
> [Purchases] - DEBUG: 💰 Purchasing product from package  - 0001 in Offering default
> [Purchases] - DEBUG: ℹ️ PaymentQueue updatedTransaction: 0001 1000000825576559 ((null)) 1000000793672260 - 1
> [Purchases] - DEBUG: ℹ️ Loaded receipt from url file:///private/var/mobile/Containers/Data/Application/2B0AF13C-FB15-47C3-9F0C-35189AACA60D/StoreKit/sandboxReceipt
> [Purchases] - DEBUG: ℹ️ Found 0 unsynced attributes for App User ID: $RCAnonymousID:f0f961e46a4d46bde0a46b17160
> [Purchases] - DEBUG: ℹ️ There are no requests currently running, starting request POST /receipts
> [Purchases] - DEBUG: ℹ️ API request started: POST /v1/receipts
> 7.3.0 - [Firebase/Analytics][I-ACS023141] Purchase is a duplicate and will not be reported. Product ID: 0001
> [connection] nw_endpoint_handler_set_adaptive_read_handler [C9.1.1 23.22.8.201:443 ready channel-flow (satisfied (Path is satisfied), viable, interface: en0, ipv4, dns)] unregister notification for read_timeout failed
> [connection] nw_endpoint_handler_set_adaptive_write_handler [C9.1.1 23.22.8.201:443 ready channel-flow (satisfied (Path is satisfied), viable, interface: en0, ipv4, dns)] unregister notification for write_timeout failed
> [Purchases] - DEBUG: ℹ️ API request completed with status: POST /v1/receipts 200
> [Purchases] - DEBUG: ℹ️ Serial request done: POST /receipts, 0 requests left in the queue
> [Purchases] - DEBUG: ℹ️ Sending updated PurchaserInfo to delegate.
> [Purchases] - DEBUG: 💰 Finishing transaction 0001 1000000825576559 (1000000793672260)
> flutter: EntitlementInfo{identifier: vip, isActive: false, willRenew: false, periodType: PeriodType.normal, latestPurchaseDate: 2021-06-11T20:25:47Z, originalPurchaseDate: 2021-03-25T15:59:11Z, expirationDate: 2021-06-11T20:30:47Z, store: Store.appStore, productIdentifier: 0001, isSandbox: true, unsubscribeDetectedAt: 2021-06-28T17:57:20Z, billingIssueDetectedAt: null}
> flutter: isLoadingSomething = true
> [Purchases] - DEBUG: ℹ️ PaymentQueue removedTransaction: 0001 1000000825576559 (1000000793672260 (null)) (null) - 1

I'm particularly intrigued with this line: `7.3.0 - [Firebase/Analytics][I-ACS023141] Purchase is a duplicate and will not be reported. Product ID: 0001`

**Comment by beylmk** [comment]
hey @dpedrinha thanks for the further info. i'm looking into this now...what seems unusual is that the `latestPurchaseDate` is 6/11, and there is `unsubscribeDetectedAt` today... 

as for the "Purchase is a duplicate" log, that's coming from Firebase...they may have some logic for determining uniqueness of a purchase. but perhaps that's a good hint that something else is up.

anyways, just wanted you to know i'm working on it! 

just to confirm -- is this happening in sandbox or production?

**Comment by beylmk** [comment]
@dpedrinha it sounds like you might have some unfinished transactions in the payment queue. assuming this is sandbox, can you try with a new sandbox account? the firebase log is making our team think some previous transaction didn't complete.

**Comment by dpedrinha** [comment]
> hey @dpedrinha thanks for the further info. i'm looking into this now...what seems unusual is that the `latestPurchaseDate` is 6/11, and there is `unsubscribeDetectedAt` today...
> 
> as for the "Purchase is a duplicate" log, that's coming from Firebase...they may have some logic for determining uniqueness of a purchase. but perhaps that's a good hint that something else is up.
> 
> anyways, just wanted you to know i'm working on it!
> 
> just to confirm -- is this happening in sandbox or production?

Firebase? I'm having this problem on ios only.

Yes, it's on sandbox.

I guess I can create a new sandbox account but it would be a real problem if this eventually happens to a user.

What I really wish I could, is to reset my purchases, attempts and users from my RC account. There are no real transactions there. It was only used for this app and it's not published yet exactly because the Apple team is also having problems going through with the purchase, thus, not approving it for the store.

Can you do it? Erase all transactions and users from my account?

**Comment by dpedrinha** [comment]
> > hey @dpedrinha thanks for the further info. i'm looking into this now...what seems unusual is that the `latestPurchaseDate` is 6/11, and there is `unsubscribeDetectedAt` today...
> > as for the "Purchase is a duplicate" log, that's coming from Firebase...they may have some logic for determining uniqueness of a purchase. but perhaps that's a good hint that something else is up.
> > anyways, just wanted you to know i'm working on it!
> > just to confirm -- is this happening in sandbox or production?
> 
> Firebase? Wow I didn't even notice this was coming from Firebase. That's weird, I'm having this problem on ios only. I guess it is just not being reported for analytics which I didn't know was reporting this.
> 
> Yes, it's on sandbox.
> 
> I guess I can create a new sandbox account but it would be a real problem if this eventually happens to a user.
> 
> What I really wish I could, is to reset my purchases, attempts and users from my RC account. There are no real transactions there. It was only used for this app and it's not published yet exactly because the Apple team is also having problems going through with the purchase, thus, not approving it for the store.
> 
> Can you do it? Erase all transactions and users from my account?

Also, if I create a new sandbox user, wouldn't it just create an alias since it's on the same device? I don't think this would work.

**Comment by rkotzy** [comment]
Hi! Deleting a customer from the dashboard does delete it from the RevenueCat database along with all it's transactions, but it does not delete the account from Apple, so if you re-open the app with the same sandbox user you'll see it be recreated in RevenueCat. When that happens, it's entire purchase history from Apple is being re-synced in the background which may cause the issue you're seeing. For example, an identical product ID that you're trying to purchase may be in the process of being synced and trigger the completion block which would explain the unexpected dates in your logs.

Creating a brand-new sandbox user and testing the purchase flow again logged in as the new account could help, let us know.

If that's the case, this issue shouldn't really be possible in production and we could chalk it up as another sandbox quirk. If you _really_ needed to use that old sandbox account for some reason, you could try re-syncing their receipt with RevenueCat through the `POST /receipt` REST API _before_ running the app: https://docs.revenuecat.com/reference#receipts

**Comment by dpedrinha** [comment]
Creating a new sandbox user didn't solve the problem.

Yesterday I created a new sandbox user, purchased a subscription, canceled it, waited for it to expire and tried purchasing it again. It worked. YAY!

So I published it to Apple Store. Today I woke up with another rejection:

> Jul 1, 2021 at 10:24 AM
> From Apple
> 
>     2. 1 Performance: App Completeness
> 
> Guideline 2.1 - Performance - App Completeness
> 
> 
> We found that your in-app purchase products exhibited one or more bugs when reviewed on iPad running iOS 14.6 on Wi-Fi.
> 
> Specifically, an error message was displayed when we attempted to subscribe.
> 
> Next Steps
> 
> When validating receipts on your server, your server needs to be able to handle a production-signed app getting its receipts from Apple’s test environment. The recommended approach is for your production server to always validate receipts against the production App Store first. If validation fails with the error code "Sandbox receipt used in production," you should validate against the test environment instead.
> 
> Resources
> 
> You can learn more about testing in-app purchase products in your development sandbox environment in App Store Connect Developer Help.
> 
> For more information on receipt validation, please see What url should I use to verify my receipt? in the In-App Purchase FAQ.
> 
> Learn how to generate a receipt validation code in App Store Connect Developer Help.
> 
> 
> 
> Please see attached screenshots for details.

So I tried making a purchase again, with the same sandbox account and it's not working anymore. Same problem as before.

One of the screenshots represent an error thrown when calling: `PurchaserInfo restoredInfo = await Purchases.restoreTransactions();`

The other one is thrown when calling: `PurchaserInfo purchaserInfo = await Purchases.purchasePackage(_offerings.current.monthly);`
I don't know the code of this error because I can't reproduce it. But I know it's not `PurchasesErrorCode.purchaseCancelledError` and not `PurchasesErrorCode.storeProblemError`.

On my device, I don't have either of this errors. I just get the behavior previously explained.

Everything works perfectly on Android.

I don't know what else to do. Any ideas?

**Comment by dpedrinha** [comment]
OK. It seems that I found the problem.

I was looking again into the example and I found one difference that shouldn't be a problem, but apparently is.

When starting the service, by calling `Purchases.setup(...)` after adding the `Purchases.addPurchaserInfoUpdateListener(...)` I was calling `Purchases.getPurchaserInfo()` to see if the user has an active purchase and automatically set the app to VIP state.

After removing it (getPurchaserInfo) the app started behaving as expected again.

But now I have a different problem: `offerings.current.monthly.product.title == ""` on iOS.

Where this data comes from? RC or Apple? Everything seems fine on appstore connect. I didn't change anything.

That's really weird.

I'll keep troubleshooting for a little while.

**Comment by rkotzy** [comment]
@dpedrinha can we keep this open issue focused on the original issue `Purchases.purchasePackage() not opening the OS purchase window after subscription expired`?

It sounds like you were able to get it to work by creating a new sandbox user? We still aren't able to reproduce this problem on our end using the Flutter sample app. If you're still facing the problem, maybe you can see if it reproduces in the Flutter sample app for you?

**Comment by dpedrinha** [comment]
> @dpedrinha can we keep this open issue focused on the original issue `Purchases.purchasePackage() not opening the OS purchase window after subscription expired`?
> 
> It sounds like you were able to get it to work by creating a new sandbox user? We still aren't able to reproduce this problem on our end using the Flutter sample app. If you're still facing the problem, maybe you can see if it reproduces in the Flutter sample app for you?

Believe me, I'm focused on it.

But if you are focusing on reproducing it on your sample, then I'm afraid you won't be of any help. If you say your sample is working, why should I bother with it? I believe you. I need to find why MY app is not working. It would be great if we, devs, could create flawless apps, wouldn't it? But the reality shows that the users will find most of the bugs. If your sample app is working it doesn't mean your whole system will work on different scenarios/apps.

I've been troubleshooting this problem and reporting my findings here, for both the RC team and other users. I even expressed my latest finding here today, which shouldn't be causing the problem, even thought it seems to be. If I'm right, I found a bug in your code.

Would you care testing it in your sample to see if you can replicate it?

**Comment by beylmk** [comment]
Hey @dpedrinha, testing in our sample app with your API key inserted would help us figure out if there's something related to your account setup (AppStore or sandbox or RevenueCat) causing this vs the actual app code.

**Comment by rkotzy** [comment]
Here's how I would approach it:
1. Does the sample app work as expected? If yes, Move to step 2. No? Provide debug logs and explain issue from sample app - since we can run the same code here hopefully can be easy to triage the problem.
2. Sample app works as expected but your app doesn't? Great! Then we know the problem is somewhere between the code in the sample and the code in your app. 
2a. Add in pieces of your code or dependencies until you can break the sample app and reproduce your same problem there. Now you've created the minimum code necessary to reproduce the issue. The should help narrow it down to where the problem exists. It may be really clear at this point what's going on and what the solution is, but if not you can share a minimum reproduction of the sample that anyone here can run on their device to see the same problem with you and help out.

Debugging apps can be an art more than a science some times. We have awesome support engineers you can always get in contact with for questions, but I'm trying to keep this Github issue focused on the problem at hand to confirm there is or isn't a bug in the Flutter SDK that might be impacting other developers.

**Comment by dpedrinha** [comment]
OK. I tried testing the sample app. But as described in #212 I'm and in [this comment](https://github.com/RevenueCat/purchases-flutter/issues/201#issuecomment-872400484) I'm now getting empty title and description for my product.

**Same behavior on the sample weather app.**

It was working fine until yesterday. Only difference is that I updated flutter today.

Related? I don't know. @rkotzy asked me to focus on the problem in the title, and I get it, but this problem didn't exist before. It appeared while trying to fix the original issue. I'm only giving as much information as I can.
